src/main/java/core/basesyntax/services/DataConverterImpl.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.services;
+
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> listOfReport) {
+        return listOfReport.stream()
+                .map(this::getFruitTransaction)
+                .toList();
+    }
+
+    private FruitTransaction getFruitTransaction(String record) {
+        FruitTransaction fruitTransaction = new FruitTransaction();
+        String[] data = record.split(",");
+        fruitTransaction.setOperation(FruitTransaction.Operation.fromCode(data[0]));
+        fruitTransaction.setFruit(data[1]);
+        fruitTransaction.setQuantity(Integer.parseInt(data[2]));
```

This violates the requirement: "The program must not ignore exceptions; instead, it must rethrow them as RuntimeException with informative messages and the original exception." There is no exception handling for parsing errors.